packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -708,13 +723,18 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
 
     node._parent = null;
     _children.remove(node);
+    for (FocusNode ancestor in ancestors) {
+      ancestor._descendants = null;
```

Comment:
Perhaps a further optimization here would be to surgically remove the descendants instead of just nuking the whole list. This is probably safer and doing the surgical removal might be more expensive than it is worth. Perhaps with a different data structure (may be set?). Anyway this is good, just something you might think about.

Reply:
Thanks, I'll give that a try and see if it's a win.

Reply:
For posterity, it's not: I tried switching to a `Set` and actively managing the set, but it was slower.